qsearch/gatesets.py

```diff
@@ -164,6 +164,74 @@ def search_layers(self, n):
         finisher = (ProductGate(NonadjacentCNOTGate(n, n-1, 0), KroneckerGate(self.single_gate, *[I]*(n-2), self.single_alt)), 1)
         return gates + [finisher]
 
+class QubitCZLinear(Gateset):
+    """A Gateset for working with CZ and single-qubit gates parameterized with U3Gate and XZXZGate on the linear topology.  This Gateset covers the same search space but uses fewer parameters than ZXZXZCNOTLinear and U3CNOTLinear."""
```

Comment:
I think the last part of this doc string is a carryover from `QubitCNOTLinear`, could you delete it? (same for below gatesets)